net-p2p/kubo/kubo-0.18.1.ebuild

```diff
@@ -0,0 +1,69 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit bash-completion-r1 go-module systemd
+
+DESCRIPTION="Main implementation of IPFS"
+HOMEPAGE="https://ipfs.io/ https://github.com/ipfs/kubo/"
+SRC_URI="https://github.com/ipfs/${PN}/releases/download/v${PV}/kubo-source.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0 BSD BSD-2 CC0-1.0 ISC MIT MPL-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="
+	acct-group/ipfs
+	acct-user/ipfs
+	sys-fs/fuse:0
+"
+RDEPEND="${DEPEND}"
+BDEPEND="dev-lang/go"
+
+DOCS=( CHANGELOG.md CONTRIBUTING.md README.md docs/ )
+
+S="${WORKDIR}"
+
+src_compile() {
+	local mygoargs
+	mygoargs=(
+		-tags release
+	)
+
+	ego build "${mygoargs[@]}" -o ipfs ./cmd/ipfs
+	ego build "${mygoargs[@]}" -o ipfswatch ./cmd/ipfswatch
+
+	IPFS_PATH="" ./ipfs commands completion bash > ipfs-completion.bash || die
+}
+
+src_test() {
+	ego test ./cmd/ipfs/... ./cmd/ipfswatch/...
+}
+
+src_install() {
+	dobin ipfs
+	dobin ipfswatch
+	newbashcomp ipfs-completion.bash ipfs
+	einstalldocs
+
+	systemd_dounit "${FILESDIR}/ipfs.service"
+	systemd_newunit "${FILESDIR}/ipfs-at.service" "ipfs@.service"
+
+	newinitd "${FILESDIR}/ipfs.init" ipfs
+	newconfd "${FILESDIR}/ipfs.confd" ipfs
+
+	keepdir /var/log/ipfs
+	fowners -R ipfs:ipfs /var/log/ipfs
+}
+
+pkg_postinst() {
+	elog 'To be able to use the ipfs service you will need to create the ipfs repository'
```

Comment:
I'm not sure if that is helpful. Given the situation that the package was installed before, and only lately updated before configured, it might make more sense to check whether a repository exists in some default location. Maybe provide a `emerge --config` option?

In that light, I'd probably drop the option to run multiple instances of ipfs (`ipfs@.service`), it doesn't make much sense anyways and is currently rather an option to run ipfs as a different user.

Reply:
I agree that sounds like a proper enhancement. Or you can use *readme.gentoo-r1.eclass* which will install the message as a file for later viewing.